app/controller/button/DigitizeButtonController.js

```diff
@@ -0,0 +1,479 @@
+     */
+    resultLayer: null,
+
+    /**
+     * Openlayers draw interaction for drawing of lines and polygons
+     */
+    drawInteraction: null,
+
+    /**
+     * Openlayers modify interaction
+     * Used in polygon draw mode only
+     */
+    modifyInteraction: null,
+
+    /**
+     * The index of the currently active group
+     * Only used when `useContextMenu` is true
+     */
+    activeGroupIdx: 0,
+
+    /**
+     * The counter reflecting the number of groups
+     * the user has created through the context menu
+     */
+    contextMenuGroupsCounter: 0,
+
+    /**
+     * Main handler which activates or deactives the interactions and listeners
+     * @param {Ext.button.Button} btn The button that has been pressed
+     * @param {boolean} pressed The toggle state of the button
+     */
+    onToggle: function(btn, pressed) {
+        var me = this;
+
+        // guess the map if not given
+        if (!me.map) {
+            me.map = BasiGX.util.Map.getMapComponent().map;
+        }
+
+        // create a temporary draw layer
+        if (!me.drawLayer) {
+            me.drawLayer = new ol.layer.Vector({
+                source: new ol.source.Vector(),
+                displayInLayerSwitcher: false
+            });
+            me.map.addLayer(me.drawLayer);
+        }
+
+        // create the draw interaction
+        if (!me.drawInteraction) {
+            var view = me.getView();
+            var type = view.getType();
+            var drawInteractionConfig = {
+                type: view.getMulti() ? 'Multi' + type : type,
+                source: me.drawLayer.getSource(),
+                condition: function(e) {
+                    // enable drawing with left mouse only
+                    return e.originalEvent && e.originalEvent.buttons === 1;
+                }
+            };
+            me.drawInteraction = new ol.interaction.Draw(drawInteractionConfig);
+            // register listeners
+            me.drawInteraction.on('drawend', me.handleDrawEnd, me);
+            me.map.addInteraction(me.drawInteraction);
+        }
+
+        // create the modify interaction
+        if (type === 'Polygon' && !me.modifyInteraction) {
+            var modifyInteractionConfig = {
+                source: me.drawLayer.getSource(),
+                deleteCondition: function(e) {
+                    return e.type === 'click' && e.originalEvent.ctrlKey;
+                }
+            };
+            me.modifyInteraction = new ol.interaction.Modify(modifyInteractionConfig);
+            me.modifyInteraction.on('modifyend', me.handleDrawEnd, me);
+            me.map.addInteraction(me.modifyInteraction);
+        }
+
+        // create a temporary result layer
+        if (!me.resultLayer) {
+            me.resultLayer = new ol.layer.Vector({
+                name: 'resultLayer',
+                displayInLayerSwitcher: false,
+                source: new ol.source.Vector(),
+                style: new ol.style.Style({
```

Comment:
Maybe it is good idea to put the style into a config / member var so the app-dev can overwrite it easily?